src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -1,771 +1,53 @@
-          }
-        }
-        .runWith(Sink.ignore)
-    }.flatten
-    ()
-  }
-
-  /** close resources for application stop */
-  private[this] def shutdown(): Unit = {
-    logger.info("Doing clean-up")
-    db.close()
-    val nodeShutdown =
-      tezosNodeOperator.node
-        .shutdown()
-        .flatMap((_: ShutdownComplete) => system.terminate())
-
-    Await.result(nodeShutdown, shutdownWait)
-    logger.info("All things closed")
-  }
-
-  /** Tries to fetch blocks head to verify if connection with Tezos node was successfully established */
-  @tailrec
-  private[this] def checkTezosConnection(): Unit =
-    Try {
-      Await.result(tezosNodeOperator.getBareBlockHead(), lorreConf.bootupConnectionCheckTimeout)
-    } match {
-      case Failure(e) =>
-        logger.error("Could not make initial connection to Tezos", e)
-        Thread.sleep(lorreConf.bootupRetryInterval.toMillis)
-        checkTezosConnection()
-      case Success(_) =>
-        logger.info("Successfully made initial connection to Tezos")
-    }
-
-  // Finds unprocessed levels for account refreshes (i.e. when there is a need to reload all accounts data from the chain)
-  private def unprocessedLevelsForRefreshingAccounts(): Future[AccountUpdatesEvents] =
-    lorreConf.chainEvents.collectFirst {
-      case ChainEvent.AccountsRefresh(levelsNeedingRefresh) if levelsNeedingRefresh.nonEmpty =>
-        db.run(TezosDb.fetchProcessedEventsLevels(ChainEvent.accountsRefresh.render)).map { levels =>
-          //used to remove processed events
-          val processed = levels.map(_.intValue).toSet
-          //we want individual event levels with the associated pattern, such that we can sort them by level
-          val unprocessedEvents = levelsNeedingRefresh.toList.flatMap {
-            case (accountPattern, levels) => levels.filterNot(processed).sorted.map(_ -> accountPattern)
-          }
-          SortedSet(unprocessedEvents: _*)
-        }
-    }.getOrElse(Future.successful(SortedSet.empty))
-
-  /** The regular loop, once connection with the node is established */
-  @tailrec
-  private[this] def mainLoop(
-      iteration: Int,
-      accountsRefreshLevels: AccountUpdatesEvents
-  ): Unit = {
-    val noOp = Future.successful(())
-    val processing = for {
-      nextRefreshes <- processAccountRefreshes(accountsRefreshLevels)
-      _ <- processTezosBlocks()
-      _ <- if (iteration % lorreConf.feeUpdateInterval == 0)
-        FeeOperations.processTezosAverageFees(lorreConf.numberOfFeesAveraged)
-      else
-        noOp
-    } yield Some(nextRefreshes)
-
-    /* Won't stop Lorre on failure from processing the chain, unless overridden by the environment to halt.
-     * Can be used to investigate issues on consistently failing block or account processing.
-     * Otherwise, any error will make Lorre proceed as expected by default (stop or wait for next cycle)
+object Lorre extends IOApp with LazyLogging with IOLogging with LorreAppConfig {
```

Comment:
Notes: cats.effect.IOApp is the functional equivalent of the traditional java Main object, where the `main(...)` method is instead defined as a `run` method that is expected to return an `IO[ExitCode]` value.

Such value describes a computation to be executed (internally by the `IOApp`) where side-effects are correctly defined inside IO values.
The value contained in the returned IO can be one of an enumeration defined in `ExitCode`'s type hierarchy.

Each corresponds roughly to a Unix process exit code (0, for success, negative to signal any form of codified error)